amazon-codeguru-reviewer-sample-app-master/src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -116,18 +116,18 @@ private void processShipmentUpdates(final LambdaLogger logger) throws Interrupte
     }
 
     private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, Map<String, Pair<Long, String>> latestStatusForTrackingNumber) {
-        final AmazonS3 s3Client = EventHandler.getS3Client();
-        logger.log("Processing Bucket: " + bucketName);
-
-        ObjectListing files = s3Client.listObjects(bucketName);
-        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>();
-
-        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) {
-            S3ObjectSummary summary = (S3ObjectSummary) iterator.next();
-            logger.log("Reading Object: " + summary.getKey());
-
-            String trackingNumber = summary.getKey().split("--")[0];
-            Pair<Long, String> lastKnownStatus = latestStatusForTrackingNumber.get(trackingNumber);
+            final AmazonS3 s3Client = EventHandler.getS3Client();
+            logger.log("Processing Bucket: " + bucketName);
+    
+            ObjectListing files = s3Client.listObjects(bucketName);
```

Comment:
### CodeGuru Findings
**Finding ARN:** APIs that are not recommended were found.
**Description:** APIs that are not recommended were found. This could indicate a deeper problem in the code.
**Severity:** High

**Finding ARN:** CWE-755 - Stack trace not included in re-thrown exception
**Description:** When re-throwing an exception, make sure to include the stack trace. Otherwise pertinent debug information is lost.
**Severity:** Medium

**Finding ARN:** Reuse AWS clients in Lambda.
**Description:** The code does not reuse clients across invocations of the Lambda function. Performance can be enhanced by using static initialization or a static constructor, global or static variables, and singletons. It allows keep alive and reuse of the HTTP connections that were established during a previous invocation.
**Severity:** Medium

**Finding ARN:** CWE-19 - Missing pagination
**Description:** Paginated results are returned instead of all results. If additional results are not checked, this might produce inaccurate results. 
**Severity:** Medium